### PR TITLE
Migrate docsearch from v2 to v3.

### PR DESCRIPTION
**Pull request by mantissahz**
Change css url and appId, apiKey.
Re-indexing the website at Algolia/Crawler website.

longhorn/longhorn#4457

Signed-off-by: James Lu <james.lu@suse.com>

### CODE REVIEW

**Comment by innobead** [comment]
@mantissahz After this, all work for non indexed versions?

**Comment by mantissahz** [comment]
>After this, all work for non indexed versions?

@innobead I still try to automatically add `version` to the index result but it would not affect the frontend codes. It would be changed at the backend of the Algolia crawler.
Now I manually add the rules for each version now such as pages under `https://longhorn.io/docs/1.3.1/` would be indexed with `version: 1.3.1` and it would be filtered by frontend with `searchParameters: { 'facetFilters': ["version:{{ $currentVersion }}"] }`

File: layouts/partials/search-bar.html [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <div class="navbar-item is-hidden-touch">
   <p class="control has-icons-left">
-    <input class="input is-rounded" type="text" placeholder="Search" id="search-input">
-    <span class="icon is-small is-left">
-      <ion-icon name="search-outline"></ion-icon>
-    </span>
+    <div class="container">
+      <div id="docsearch"></div>
```
Comment (innobead) [inline layouts/partials/search-bar.html:4]: I guess this change is from the their example? But will this impact CSS styling change?

**Comment by innobead** [comment]
> >After this, all work for non indexed versions?
> 
> 
> 
> @innobead I still try to automatically add `version` to the index result but it would not affect the frontend codes. It would be changed at the backend of the Algolia crawler.
> 
> Now I manually add the rules for each version now such as pages under `https://longhorn.io/docs/1.3.1/` would be indexed with `version: 1.3.1` and it would be filtered by frontend with `searchParameters: { 'facetFilters': ["version:{{ $currentVersion }}"] }`

IIRC, this is what other versions work after Jenting change, correct?

Anything new changed? Or anything I probably missed?

File: layouts/partials/search-bar.html (hunk 1 above)
Comment (mantissahz) [reply layouts/partials/search-bar.html:4]: Yeah, It's from Docserach official website.
I am not good at web UI and need someone's double check, but it seams the deploy-preview looks the same as original one except the style of search bar.

Comment (innobead) [reply layouts/partials/search-bar.html:4]: You can run `make serve` to see locally, but I believe the style will be impacted. Or, we can check it together tomorrow.

**Comment by mantissahz** [comment]
Just changed the backend config, and I think it is ready for all and non indexed versions.

File: layouts/partials/search-bar.html (hunk 1 above)
Comment (innobead) [reply layouts/partials/search-bar.html:4]: After discussing, the style will be controlled by docsearch styling instead, so we don't need to worry about it.

**Review by innobead — APPROVED** [review]

File: layouts/partials/search-bar.html (hunk 1 above)
Comment (mantissahz) [reply layouts/partials/search-bar.html]: Yeah, I did use `make serve` to check the changes.
Modified. Thanks. only `<div id="docsearch"></div>`

**Pull request merged**